src/osd/PGLog.cc

```diff
@@ -131,10 +131,18 @@ void PGLog::IndexedLog::trim(
     }
   }
 
-  while (!dups.empty()) {
+  // we can hit an inflated `dups` b/c of https://tracker.ceph.com/issues/53729
```

Comment:
this approach looks good to me!

should update the osd_pg_log_dups_tracked docs since the meaning of the option is a bit different now - we'll have either that many log entries, or if max log entries is lower, that many dup entries in addition to log entries.